### PR TITLE
Refactor: Add types to categorizeContractCallTxObject util function

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/popup/utils/utils.ts [hunk 1]
```diff
@@ -190,3 +192,48 @@ export function compareCaseInsensitive(
...
+export function categorizeContractCallTxObject(transaction: ITransaction): {
+  amount: number
+  to?: string
+  token: string
+  url?: string
+  note?: string
+} | null {
+  if (transaction.tx.type?.toLowerCase() !== SCHEMA.TX_TYPE.contractCall.toLowerCase()) {
+    return null;
+  }
```
Comment (pstachel) [inline src/popup/utils/utils.ts:205]: check out `compareCaseInsensitive` function

**Review by pstachel — APPROVED** [review]

**Review by CedrikNikita — CHANGES_REQUESTED** [review]

**Review by konradodo — APPROVED** [review]

File: src/popup/utils/utils.ts [hunk 2]
```diff
@@ -188,5 +190,50 @@ export function compareCaseInsensitive(
...
+  token: string
+  url?: string
+  note?: string
+} | null {
+  if (!compareCaseInsensitive(transaction.tx.type, SCHEMA.TX_TYPE.contractCall)) {
+    return null;
+  }
+  if (transaction.incomplete || transaction.pending) {
+    return {
+      amount: transaction.tx.amount,
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/popup/utils/utils.ts]: OK, fixed. It turns out that this function can receive both `ITransaction` and `IPendingTransaction`. I needed to align some types to handle this situation. Please see now.

File: src/popup/components/TransferReview.vue [hunk 3]
```diff
@@ -224,8 +224,9 @@ export default defineComponent({
             amount,
             recipient,
             hash: actionResult.hash,
-            type: 'spendToken',
             pendingTokenTx: true,
+            pending: true,
```
Comment (CedrikNikita) [inline src/popup/components/TransferReview.vue:228]: `pending` field is being added by the `addPendingTransaction` action itselft

Comment (peronczyk) [reply src/popup/components/TransferReview.vue:228]: `addPendingTransaction` is a store function, so there's no types there. I added it here as the `pending` property is required by the `IPendingTransaction`. In the future, when we will start to remove the Vuex we can try to do it better.

**Review by CedrikNikita — APPROVED** [review]
Code changes look reasonable. Did not test it.

**Pull request merged**